cocos/render-scene/core/pass.ts

```diff
@@ -523,7 +561,16 @@ export class Pass {
             this._defines[patch.name] = patch.value;
         }
 
-        const shader = programLib.getGFXShader(this._device, this._programName, this._defines, pipeline);
+        let shader: Shader | null = null;
+        if (cclegacy.rendering && cclegacy.rendering.enableEffectImport) {
+            const program = (cclegacy.rendering.programLib as ProgramLibrary)
```

Comment:
Why need to use `as`?

Reply:
When using cclegacy, all members become `any`.
Here we do an explicit cast, to ensure type-safeness. So I can pass function parameters correctly, with type checking.

Reply:
Got it. As cclegacy is deprecated. Is there any other way to use rendering?

Reply:
I don't know yet. It depends on module system.

Reply:
Ok.